platform/ios/src/MGLMapView.mm

```diff
@@ -1615,6 +1656,12 @@ - (void)handleTwoFingerDragGesture:(UIPanGestureRecognizer *)twoFingerDrag
         [self notifyGestureDidEndWithDrift:NO];
         [self unrotateIfNeededForGesture];
     }
+    
+    if ([self.delegate respondsToSelector:@selector(mapView:shouldChangeFromCamera:toCamera:)]
```

Comment:
Unless you check the gesture recognizer's state (or didChangeCamera), the camera will be reset to oldCamera before the gesture is complete.